ProfitSite/src/main/java/Profit/AdList.java

```diff
@@ -0,0 +1,170 @@
+package Profit;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class AdList {
+    private List<AdItem> ads;
+
+    private static class SortByDescendingDate implements Comparator<AdItem> {
+        public int compare(AdItem a, AdItem b) {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            Date date1 = new Date();
+            Date date2 = new Date();
+            try {
+                date1 = sdf.parse(a.getCreatedAt());
```

Comment:
Why parsing again? You could use `getDateCreateAt()` )
But I strictly recommend that you just store dates as `Date`s from the start, and not at all strings